packages/design-tokens/design-token-previews/components/IconSizePreview.tsx

```diff
@@ -0,0 +1,19 @@
+import React from "react";
+import { SystemProp, Theme } from "@xstyled/styled-components";
+import { Icon } from "../../../components/src/components/Icon/Icon";
+import { ThemeProvider, theme } from "../../../theme";
+
+const IconSizePreview = ({ pixels }: { pixels: string }): JSX.Element => {
+  return (
+    <ThemeProvider theme={theme}>
+      <Icon
+        decorative
+        display="flex"
+        icon="HeartIcon"
+        size={pixels as SystemProp<keyof Theme["sizes"], Theme>}
+      />
+    </ThemeProvider>
+  );
+};
```

Comment:
Following this suggestion, you don't need to have it specifically for size and you could also set the other props externally. The only reason to have this component IMO is only the `ThemeProvider`. 

```suggestion
const IconPreview = (props: IconProps): JSX.Element => {
  return (
    <ThemeProvider theme={theme}>
      <Icon
        {...props}
        decorative
        display="flex"
        icon="HeartIcon"
      />
    </ThemeProvider>
  );
};
```

Another thing to consider is, as we already have the `createIconSizePreview` function, remove this component and return the `ThemeProvider` + `Icon` there. Which seems better to me because the function is small and we are not reusing this component anywhere.

Reply:
I tried to implement this with Stephan and we couldn't figure out what to do here. In this case, it's working so I wouldn't change it.